homeassistant/components/sensor/__init__.py

```diff
@@ -786,16 +786,8 @@ def _update_suggested_precision(self) -> None:
             ratio_log = floor(ratio_log) if ratio_log > 0 else ceil(ratio_log)
             display_precision = max(0, display_precision + ratio_log)
 
-        if display_precision is None and (
-            DOMAIN not in self.registry_entry.options
-            or "suggested_display_precision" not in self.registry_entry.options
-        ):
-            return
         sensor_options: Mapping[str, Any] = self.registry_entry.options.get(DOMAIN, {})
-        if (
-            "suggested_display_precision" in sensor_options
-            and sensor_options["suggested_display_precision"] == display_precision
-        ):
+        if sensor_options.get("suggested_display_precision") == display_precision:
```

Comment:
I think we should keep the in-check here:
```suggestion
        if (
            "suggested_display_precision" in sensor_options
            and sensor_options["suggested_display_precision"] == display_precision
        ):
```

Reply:
I guess it's more explicit this way, but if it isn't in sensor_options, the get will return None, which won't match display_precision, so the end result should be the same either way.

Reply:
With your version an absence of the key is treated the same as the key being present and set to `None`.